rero_ils/modules/notifications/subclasses/reminder.py

```diff
@@ -0,0 +1,173 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019 RERO
+# Copyright (C) 2020 UCLouvain
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""API for manipulating reminder circulation notifications."""
+
+from __future__ import absolute_import, print_function
+
+import ciso8601
+from num2words import num2words
+
+from rero_ils.modules.circ_policies.api import DUE_SOON_REMINDER_TYPE, \
+    OVERDUE_REMINDER_TYPE
+from rero_ils.modules.documents.dumpers import DocumentNotificationDumper
+from rero_ils.modules.libraries.dumpers import \
+    LibraryCirculationNotificationDumper
+from rero_ils.modules.loans.utils import get_circ_policy
+from rero_ils.modules.patrons.dumpers import PatronNotificationDumper
+from rero_ils.utils import language_iso639_2to1
+
+from .circulation import CirculationNotification
+from ..api import NotificationsSearch
+from ..models import NotificationChannel, NotificationType
+
+
+class ReminderCirculationNotification(CirculationNotification):
+    """Reminder circulation notifications class.
+
+    A reminder notification is a message send to a patron to notify that a
+    borrowed document should requires its attention.
+
+    The reminder notification set groups DUE_SOON and OVERDUE notifications.
+    The settings about this notification set could be found into the
+    circulation policy (CIPO) related to the loan. At this time.
+    """
+
+    @property
+    def _cipo_reminder(self):
+        """Get the related circulation policy reminder for this notification.
+
+        REMINDERS notification are linked to setting define into the
+        circulation policy related to the loan. The cipo reminders could not
+        exists (because the item circulation category [itty] changed, the
+        patron type changed, ... from the notification creation.
```

Comment:
```suggestion
        REMINDERS notification are linked to settings defined in the
        circulation policy related to the loan. The cipo reminders may not
        exists (because the item circulation category [itty] changed, the
        patron type changed… since the notification creation.
```